packages/zoe/README.md

```diff
@@ -124,3 +132,25 @@ harden(prepare);
 ```
 
 For an example contract upgrade, see the test at https://github.com/Agoric/agoric-sdk/blob/master/packages/zoe/test/swingsetTests/upgradeCoveredCall/test-coveredCall-service-upgrade.js .
+
+### Durability
+
+The contract must retain in durable storage anything that must persist between incarnations. All other state will be lost.
+
+### Kinds
+
+The contract defines the kinds that are held in durable storage. Thus the function calls that define the kinds must be run before the objects are deserialized from durable storage.
+
+# Crank
+
+After the first incarnation, `prepare` must return in one crank. Therefore any
```

Comment:
This is wrong. For any incarnation, a `prepare` for a new instance or new class can happen in any crank, without restriction. What must happen in the first crank is a `prepare` of Exo instances or ExoClasses that were already defined in a previous incarnation. In this regard, the only thing unique about the first incarnation is that there were no previous incarnations, and so no Exo instances or ExoClasses that were previously defined. Thus no `prepare`s that must happen during the first crank.